src/alchemlyb/parsing/lammps.py

```diff
@@ -0,0 +1,1303 @@
+
+    Returns
+    -------
+    lambda_values : list
+        List of tuples lambda values contained in the file.
+    lambda_pairs : list
+        List of tuples containing two floats, lambda and lambda'.
+    lambda2 : float
+        Value of lambda2 that is held constant.
+
+    .. versionadded:: 2.5.0
+
+    """
+
+    lambda_pairs = [
+        _tuple_from_filename(y, indices=indices, prec=prec) for y in fep_files
+    ]
+    if len(indices) == 3:
+        lambda2 = list(
+            set(
+                [
+                    _lambda_from_filename(y, index=indices[2], prec=prec)
+                    for y in fep_files
+                ]
+            )
+        )
+        if len(lambda2) > 1:
+            raise ValueError(
+                "More than one value of lambda2 is present in the provided files."
+                f" Restrict filename input to one of: {lambda2}"
+            )
+        else:
+            lambda2 = lambda2[0]
+    else:
+        lambda2 = None
+
+    lambda_values = sorted(list(set([x for y in lambda_pairs for x in y])))
+
+    if [x for x in lambda_values if round(float(x), prec) < 0]:
+        raise ValueError("Lambda values must be positive: {}".format(lambda_values))
+
+    # check that all needed lamba combinations are present
+    lambda_dict = {x: [y[1] for y in lambda_pairs if y[0] == x] for x in lambda_values}
+
+    # Check for MBAR content
+    missing_combinations_mbar = []
+    for lambda_value, lambda_array in lambda_dict.items():
+        missing_combinations_mbar.extend(
+            [(lambda_value, x) for x in lambda_values if x not in lambda_array]
+        )
+
+    if missing_combinations_mbar:
+        warnings.warn(
+            "The following combinations of lambda and lambda prime are missing for MBAR analysis: {}".format(
+                missing_combinations_mbar
+            )
+        )
+    else:
+        return lambda_values, lambda_pairs, lambda2
+
+    # Check for BAR content
+    missing_combinations_bar = []
+    lambda_values.sort()
+    for ind, (lambda_value, lambda_array) in enumerate(lambda_dict.items()):
+        if ind == 0:
+            tmp_array = [lambda_values[ind], lambda_values[ind + 1]]
+        elif ind == len(lambda_dict) - 1:
+            tmp_array = [lambda_values[ind - 1], lambda_values[ind]]
+        else:
+            tmp_array = [
+                lambda_values[ind - 1],
+                lambda_values[ind],
+                lambda_values[ind + 1],
+            ]
+
+        missing_combinations_bar.extend(
+            [(lambda_value, x) for x in tmp_array if x not in lambda_array]
+        )
+
+    if missing_combinations_bar and not force:
+        raise ValueError(
+            "BAR calculation cannot be performed without the following lambda-lambda prime combinations: {}".format(
+                missing_combinations_bar
+            )
+        )
+
+    return lambda_values, lambda_pairs, lambda2
+
+
+@_init_attrs
+def extract_u_nk_from_u_n(
```

Comment:
You might want to add a doc page saying when do you need extract_u_nk_from_u_n instead of extract_u_nk